prov/shm/src/smr_progress.c

```diff
@@ -201,9 +201,8 @@ static void smr_progress_resp(struct smr_ep *ep)
 	int ret;
 
 	pthread_spin_lock(&ep->region->lock);
-	ofi_genlock_lock(&ep->util_ep.tx_cq->cq_lock);
-	while (!ofi_cirque_isempty(smr_resp_queue(ep->region)) &&
-	       !ofi_cirque_isfull(ep->util_ep.tx_cq->cirq)) {
+	ofi_mutex_lock(&ep->util_ep.lock);
+	while (!ofi_cirque_isempty(smr_resp_queue(ep->region))) {
```

Comment:
This also loses the ofi_cirque_isfull check.  Do the changes now allow overrunning the tx cq?